rand_distr/src/normal_inverse_gaussian.rs

```diff
@@ -0,0 +1,107 @@
+use crate::{Distribution, Float, InverseGaussian, Standard, StandardNormal};
+use rand::Rng;
+
+/// Error type returned from `NormalInverseGaussian::new`
+#[derive(Debug, PartialEq)]
+pub enum Error {
+    /// `alpha <= 0` or `nan`.
+    AlphaNegativeOrNull,
+    /// `|beta| >= alpha` or `nan`.
+    AbsoluteBetaLessThanAlpha,
+}
+
+/// The [normal-inverse Gaussian distribution](https://en.wikipedia.org/wiki/Normal-inverse_Gaussian_distribution)
+#[derive(Debug)]
+pub struct NormalInverseGaussian<N> {
+    alpha: N,
```

Comment:
Looks like there's no need to store α?

Reply:
No there's not but I kept it because of the `Debug` derivation, to be able to see the distribution parameters when printing it.
I can remove it if you prefer.